src/analyze/analyze.c

```diff
@@ -1212,7 +1256,75 @@ static int dot(int argc, char *argv[], void *userdata) {
+                { "Style", "Requisite",         config_parse_string,  0, &dependency_style[UNIT_REQUISITE] },
+                { "Style", "Wants",             config_parse_string,  0, &dependency_style[UNIT_WANTS]     },
+                { "Style", "Conflicts",         config_parse_string,  0, &dependency_style[UNIT_CONFLICTS] },
+                { "Style", "BindsTo",           config_parse_string,  0, &dependency_style[UNIT_BINDS_TO]  },
+                { "Style", "PartOf",            config_parse_string,  0, &dependency_style[UNIT_PART_OF]   },
+                { "Style", "OnFailure",         config_parse_string,  0, &dependency_style[UNIT_ON_FAILURE]},
+                { "Style", "Triggers",          config_parse_string,  0, &dependency_style[UNIT_TRIGGERS]  },
+                {}
+        };
+
+        static const char* colors[] = {
+                "black",    /* UNIT_REQUIRES */
+                "blue",     /* UNIT_REQUISITE */
+                "grey66",   /* UNIT_WANTS */
+                "magenta",  /* UNIT_BINDS_TO */
+                "black",    /* UNIT_PART_OF */
+                "",
+                "",
+                "",
+                "",
+                "",
+                "red",      /* UNIT_CONFLICTS */
+                "",
+                "",
+                "green",    /* UNIT_AFTER */
+                "orange",   /* UNIT_ON_FAILURE */
+                "grey66"    /* UNIT_TRIGGERS */
+        };
+
+        for (int dependency = UNIT_REQUIRES; dependency <= UNIT_TRIGGERS; ++dependency) {
+                dependency_color[dependency] = strdup(colors[dependency]);
+                if (strcmp(colors[dependency], "") != 0) {
+                        if (dependency == UNIT_PART_OF)
+                                dependency_style[dependency] = strdup("dashed");
+                        else if (dependency == UNIT_TRIGGERS)
+                                dependency_style[dependency] = strdup("dotted");
+                        else
+                                dependency_style[dependency] = strdup("solid");
+                }
+        }
+
+        if (!dependency_color[UNIT_AFTER]      || !dependency_style[UNIT_AFTER]      ||
+            !dependency_color[UNIT_REQUIRES]   || !dependency_style[UNIT_REQUIRES]   ||
+            !dependency_color[UNIT_REQUISITE]  || !dependency_style[UNIT_REQUISITE]  ||
+            !dependency_color[UNIT_WANTS]      || !dependency_style[UNIT_WANTS]      ||
+            !dependency_color[UNIT_CONFLICTS]  || !dependency_style[UNIT_CONFLICTS]  ||
+            !dependency_color[UNIT_BINDS_TO]   || !dependency_style[UNIT_BINDS_TO]   ||
+            !dependency_color[UNIT_PART_OF]    || !dependency_style[UNIT_PART_OF]    ||
+            !dependency_color[UNIT_ON_FAILURE] || !dependency_style[UNIT_ON_FAILURE] ||
+            !dependency_color[UNIT_TRIGGERS]   || !dependency_style[UNIT_TRIGGERS]
+        )
+            return log_oom();
```

Comment:
please do the oom check in the loop, instead of listig every dep here in a huge if statement

Reply:
done in the loop